File: Ruby/hisab_kitab.rb
input_file = ARGV.first

input = File.open(input_file)

individuals = {}
toPay = {}
total = 0

while(line = input.gets) 
	arr = line.split(' - ');
	individuals[arr[0]] = individuals[arr[0]].to_i() + arr[1].to_i() 
end

len = individuals.length
if len <= 0
	puts '#' * 25
	puts "INVALID INPUT FILE"
	puts '#' * 25
	Process.exit(0)
end

puts '-' * 25
puts "Individual Kharcha :"
puts '-' * 25


for name, amount in individuals
	puts "%s : Rs. %s" % [name, amount]
	total += amount
end

puts '-' * 25
puts "Total Kharcha : Rs. #{total}"
puts '-' * 25
puts "Each one to pay :"
puts '-' * 25


total = total * 1.0 / len

for name, amount in individuals
	toPay[name] = total - amount
	toPay[name] = sprintf("%.0f", toPay[name])
	puts "%s : Rs. %s" % [name, toPay[name]]
end
puts '-' * 25


input.close

File: Perl/hisab_kitab.pl
#!/usr/bin/perl

use strict;
use warnings;

my $individuals;
my $toPay;
my $total;

while(<>) {
	chomp;
	next unless validateInput($_);
	my @spl = split (' - ');
	$individuals->{$spl[0]} += $spl[1]; 
}

printDash();
print "Individual Kharcha :\n";
printDash();

for my $k (keys %$individuals) {
	print "$k : Rs. $individuals->{$k}\n";
	$total += $individuals->{$k};
}

printDash();
print "Total Kharcha : Rs. $total\n";
printDash();
print "Each one to pay : \n";
printDash();

$total /= 3;
for my $k (keys %$individuals) {
	$toPay->{$k} = $total - $individuals->{$k};
	print "$k : Rs. ", sprintf("%.0f", $toPay->{$k}), "\n";
}
printDash();

sub printDash {
	print "-------------------------\n";
}

sub validateInput {
	my $input = shift;

	return 1 if $input =~ m/^\w+? - \d+?(?: - .*)?$/;

	warn("Validation not passed : $input. Ignoring.");
	return 0;
}
